.github/workflows/test.yml

```diff
@@ -0,0 +1,68 @@
+name: Checks
+
+on: [push, pull_request]
+
+jobs:
+  checks:
+    timeout-minutes: 45
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macos-11]
+      fail-fast: false
+
+    steps:
+      - name: Fetch source code
+        uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            .cargo/bin/
+            .cargo/registry/index/
+            .cargo/registry/cache/
+            .cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
this repo has the rather anomalous behavior of not checking the `Cargo.lock` file into git, which as I understand it has its reasons, but from my experience has had some real downsides

Reply:
in my other one, I use `Cargo.toml`, while realizing that that's not best practice

Reply:
> this repo has the rather anomalous behavior of not checking the `Cargo.lock` file into git, which as I understand it has its reasons, but from my experience has had some real downsides

Hmm I think this is actually best practice for libraries (but not binaries) — https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Reply:
> in my other one, I use `Cargo.toml`, while realizing that that's not best practice

Thanks, should update

Reply:
That kinda makes sense. The zomes themselves are more like binaries, while the things in `lib` are libraries, so 🤷 I'm not sure

Reply:
> That kinda makes sense. The zomes themselves are more like binaries, while the things in `lib` are libraries, so 🤷 I'm not sure

More nuanced suggestion here: https://stackoverflow.com/a/63727523